staging/src/k8s.io/legacy-cloud-providers/aws/aws_routes.go

```diff
@@ -127,13 +127,40 @@ func (c *Cloud) ListRoutes(ctx context.Context, clusterName string) ([]*cloudpro
 	return routes, nil
 }
 
-// Sets the instance attribute "source-dest-check" to the specified value
-func (c *Cloud) configureInstanceSourceDestCheck(instanceID string, sourceDestCheck bool) error {
-	request := &ec2.ModifyInstanceAttributeInput{}
-	request.InstanceId = aws.String(instanceID)
-	request.SourceDestCheck = &ec2.AttributeBooleanValue{Value: aws.Bool(sourceDestCheck)}
-
-	_, err := c.ec2.ModifyInstanceAttribute(request)
+// Sets the instance's primary network interface attribute "source-dest-check" to the specified value
+func (c *Cloud) configureInstanceSourceDestCheck(instance *ec2.Instance, sourceDestCheck bool) error {
+	instanceID := aws.StringValue(instance.InstanceId)
+	srcDstCheck := &ec2.AttributeBooleanValue{Value: aws.Bool(sourceDestCheck)}
+
+	var err error
+	switch len(instance.NetworkInterfaces) {
+	case 0:
+		return fmt.Errorf("no network interface found on instance %s", instanceID)
+	case 1:
+		_, err = c.ec2.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
+			InstanceId:      instance.InstanceId,
+			SourceDestCheck: srcDstCheck,
+		})
+	default:
+		// If there are multiple network interfaces attached to the instance,
+		// the invocation of `ModifyInstanceAttributes()` would fail with the following error:
+		// "InvalidInstanceID: There are multiple interfaces attached to instance 'i-xx'. Please specify an interface ID for the operation instead."
+		// So we need to modify the attribute of primary network interface here.
+		var primaryInterfaceID *string
+		for _, device := range instance.NetworkInterfaces {
+			if aws.Int64Value(device.Attachment.DeviceIndex) == 0 {
```

Comment:
What's the reason for choosing the network interface with device index 0?

Reply:
as per https://github.com/terraform-providers/terraform-provider-aws/blob/f213ed9afac0fa648c915969d571114636660c14/aws/resource_aws_instance.go#L1135-L1140

Reply:
also from the official doc: https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#InstanceNetworkInterfaceSpecification

```
    // The position of the network interface in the attachment order. A primary
    // network interface has a device index of 0.
    //
    // If you specify a network interface when launching an instance, you must specify
    // the device index.
    DeviceIndex *int64 `locationName:"deviceIndex" type:"integer"`
```